src/Aperture/elliptical.jl

```diff
@@ -0,0 +1,89 @@
+export EllipticalAperture,
+       EllipticalAnnulus
+
+"""
+   EllipticalAperture(x, y, a, b, theta)
+
+A elliptical aperture.
+
+Where a >= b > 0 and -pi/2 <= theta <= pi/2
+
+# Examples
+```jldoctest
+julia> EllipticalAperture(2,2,2,2,1.2)
+EllipticalAperture(2.0, 2.0, a=2.0, b=2.0, theta=1.2)
+
+```
+"""
+struct EllipticalAperture{T <: Number} <: AbstractAperture
+    x::T
+    y::T
+    a::T
+    b::T
+    theta::T
+    cxx::T
+    cxy::T
+    cyy::T
+end
+
+function EllipticalAperture(x, y, a, b, theta)
+    if (b < 0.0 || a < b || theta < -pi/2 || theta > pi/2)
+            error("illegal ellipse parameters. " * "Require a >= b > 0.0, -pi/2 <= theta <= pi/2")
+    end
```

Comment:
I would suggest taking `theta` as degrees since that is much easier for a human to visualize and input. 

Also, there should be no need to explicitly reject `theta` outside some range- we can just `mod(theta, 360)` at most, but at the end of the day when we put it into `sin` or `cos` the trigonometry will be the same if its 20deg vs 380deg vs -340deg.

Also, why use `"" * ""` instead of just one string?